src/platform/Zephyr/WiFiManager.h

```diff
@@ -39,11 +43,34 @@ class WiFiManager
 public:
     enum class StationStatus : uint8_t
     {
+        NONE,
         DISCONNECTED,
+        DISABLED,
         SCANNING,
-        COMPLETED,
-        CONNECTED = COMPLETED,
-        NONE
+        CONNECTING,
+        CONNECTED,
+        PROVISIONING,
+        FULLY_PROVISIONED
+    };
+
+    class StatusMap
+    {
+    public:
+        static StatusMap & GetMap()
+        {
+            static StatusMap sInstance;
+            return sInstance;
+        }
+        StationStatus operator[](enum wpa_states aWpaState);
```

Comment:
I don't know if this class will get extended... well I wanted to emulate a map concept in simplistic way and having it as a class allows for operator[], which comes handy, to be easily overload. I am not sure if this will still be needed if the callback API is ready. Can we keep it this way and reconsider during refactoring, which will eventually happen?

Reply:
Yes, we can!